### DISCUSSION TITLE
[Feature] speedup heatmap view in map display

### DISCUSSION THREAD

**Discussion by Frava735** [immich-app/immich#11977] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be nice to find ways (new parameters) to speedup hestmap build when using map display in android mobile app.
Example: allowing to default switch to map display with bubbles with picture numbers in corresponding areas, based on zoom. Larger zoom would load up almost instantly. 

Or allow in android app user settings to use by default  map display like the one in web app.

[Note: I am using a recent fast enough smartphone,  Wi-Fi network is 1Gbps, immich server is on a 1/2.5Gpbs wired network adapter,  immich server HW is powerful enough].

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by radh21301** [comment]
I agree. It definitely needs a lot of optimization. It currently is a pain to use(I have 40,000 assets- using via local IP) - takes about 10 seconds for an initial load, then subsequently additional 5 seconds after every zoom - Till I get to a small enough area. Moreover, once you zoom in, you cannot zoom in further till whatever processing it does to fetch the assets in the current zoom level is completed. I would expect it to stop fetching the assets when I the to change the zoom level further - and allow me to zoom. But nope, got to wait for that time when the screen is basically frozen.

Overall, it takes me about 20 seconds to go to the area I want. Definitely not a pleasant experience.

**Reply by Frava735** [reply]
I agree.
The part we should emphasize on is maybe the "frozen screen" we get.
I becomes really painful to use when you have 40k+ pictures (I am close to 80k, it takes ages to refresh...). 
I don't understand why the Android app didn't get the same map/treatment as the web app, which is definitely fast enough.

**Reply by alextran1502** [reply]
> I don't understand why the Android app didn't get the same map/treatment as the web app, which is definitely fast enough.

@Frava735 Two different code bases, two different libraries, and two different rendering methods. And I am pretty sure we currently use the default rendering methods on each.

**Comment by ldivl** [comment]
The map doesn't work on IOS either.   I have more like 180k photos though, and it just perpetually freezes on IOS.   I really don't use the feature often though.

**Comment by ffchung** [comment]
Same here, iOS Ver 1.123.0, The map view will freeze for a mins for every move. It working well with Web Version.

**Comment by atle85** [comment]
Any updates here?

**Reply by alextran1502** [reply]
There is not
